### PR TITLE
Add additional date validations for TSP edit form

### PR DESCRIPTION
**Pull request by LeDeep**
## Description

There were a few issues with date selection validation in the TSP SIT edit form. See here for context: https://www.pivotaltracker.com/story/show/166049743 

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

Make a `DESTINATION` SIT request. Get to the point where your destination SIT is in `DELIVERED` state with a date out. Try changing the actual start date by manually typing in a date that is after the date out. You will see a validation that the date must be no later than the date out date, and the save button will be disabled.

Next try manually changing the out date by typing a date that is before the actual start date. You will see a validation that the date must be no earlier than the actual start date, and the save button will be disabled.

The date out and actual start date calendars work together, in that an update to one will also update the other, without a refresh or save. So you no longer have the issue where you pick an actual start date and then choose a date out that is before that date.
Select an 'ActualStartDate'. Open the `DateOut` calendar and you will see that all dates before the `ActualStartDate` you selected are disabled.

Select an `OutDate`. Open the `ActualStartDate` calendar and you will see that all dates after the `DateOut` you selected are disabled.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/166049743) for this change

## Screenshots

Screen Shot 2019-07-25 at 3 42 26 PM

Screen Shot 2019-07-30 at 3 46 15 PM

### CODE REVIEW

File: src/shared/JsonSchemaForm/JsonSchemaField.js [hunk 1]
```diff
@@ -359,7 +363,7 @@ const createSchemaField = (
     const fieldFormat = swaggerField.format;
     if (fieldFormat === 'date' && !isNil(minDate)) {
```
Comment (reggieriser) [inline src/shared/JsonSchemaForm/JsonSchemaField.js]: Could we have a restricted date field that set a `maxDate` but not a `minDate`?  It looks like we assume here that we always have a `minDate` if we want the validation checks?

Comment (LeDeep) [reply src/shared/JsonSchemaForm/JsonSchemaField.js]: Currently, this is the only use of `maxDate`. It's possible that we could run into this issue in the future. I can add an `or` check here to safeguard against a potential failed validation if `maxDate` is used without `minDate`.

File: src/shared/JsonSchemaForm/JsonSchemaField.js [hunk 2]
```diff
@@ -359,9 +363,9 @@ const createSchemaField = (
     }
   } else if (swaggerField.type === 'string') {
     const fieldFormat = swaggerField.format;
-    if (fieldFormat === 'date' && !isNil(minDate)) {
+    if ((fieldFormat === 'date' && !isNil(minDate)) || !isNil(maxDate)) {
```
Comment (reggieriser) [inline src/shared/JsonSchemaForm/JsonSchemaField.js]: Are the parens right here?  Don't you want `if (fieldFormat === 'date' && (!isNil(minDate) || !isNil(maxDate)))`?  In other words, is this a date field and (has either a minDate or a maxDate)?

Comment (LeDeep) [reply src/shared/JsonSchemaForm/JsonSchemaField.js]: What you wrote is what I had. When I went to commit, the linter removed the outside parens. It still worked when I tested, so chose not to override the linter.

Comment (LeDeep) [reply src/shared/JsonSchemaForm/JsonSchemaField.js]: Fixed. Realized I had an extra set of parens that was causing the issue.

**Pull request merged**